### PR TITLE
fix for events incremental

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**
#45 #48 

**This PR will result in the following new package version:**

v0.13.0

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Breaking Changes (`--full-refresh` required after upgrading)  
- Added a field called `first_open_or_click_event_at` in the `iterable__user_campaign` model. This timestamp shows the first time a user interacted with a campaign, recording the earliest occurring event out of 'emailOpen', 'emailClick', and 'pushOpen'. ([PR #50](https://github.com/fivetran/dbt_iterable/pull/50))
- Corrected the incremental filter in `iterable__events` model to now use the `created_on` date field instead of the `created_at` timestamp. 
   - Previously, this would potentially exclude late-arriving new records from populating in the end models if they had an older `created_at` value than what was present in the model. Switching to `created_on` widens the criteria.
- Updated upstream `stg_iterable__user_history` model from materializing as a table to a view in order to improve performance.
- In order to ensure no issues a `--full-refresh` is required after upgrading.

## Under the Hood
- In addition to using `created_on` in the incremental logic in `iterable__events`, we introduced a `iterable_lookback_window` variable to increase the window for accommodating potential late-arriving records. The default is 7 days prior to the maximum `created_on` value present in the `iterable__events` model, but you may customize this by setting the var `iterable_lookback_window ` in your dbt_project.yml. See the [Lookback Window section of the README](https://github.com/fivetran/dbt_iterable/blob/main/README.md#lookback-window) for more details.
- Added a section in the [README]((https://github.com/fivetran/dbt_iterable/blob/main/README.md#pivoting-out-event-metrics)) about the `iterable__event_metrics` variable and how to use it to specify which event metrics to pivot out.
- Removes `created_on` from the uniqueness test in `iterable__events`. Uniqueness is now tested solely on `unique_event_id`, a surrogate key made up of `event_id` (`_fivetran_id` in the raw table, which is a Fivetran-created unique identifier derived from hashing campaign_id, created_at, and event_name) and `_fivetran_user_id` (a Fivetran-created column derived from a hash of `user_id` and/or `email`).
- Modified the `event` seed data to more accurately represent real-life data, with a unique `_fivetran_id` for each campaign_id, created_at, and event_name.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [na] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- see internal ticket thread for validation in addition to the below:
image

This error in `consistency_user_campaign `  is expected due to the addition of a new field `first_open_or_click_event_at` in this branch. 

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: models/iterable__events.sql
Comment (fivetran-joemarkiewicz) [inline models/iterable__events.sql:20]: Also small request, but can you order the parameters in the same order they are when defining the macro. So it should go: (`from_date`, `datepart`, `interval`)

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli thanks for working through this. This is looking really good I just have a few comments and suggestions before approving. Also, would you please be able to run the validation tests for this package and share the results in the PR description to confirm there are no unexpected changes between dev and prod.

File: models/iterable__events.sql
Comment (fivetran-reneeli) [reply models/iterable__events.sql:20]: done!

Comment (fivetran-reneeli) [reply models/iterable__events.sql:20]: will do!

File: README.md
Comment (fivetran-reneeli) [reply README.md:1]: oh good point! merging it in

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:8]: updated

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:12]: updated along with contributors issues + PRs!

File: README.md
Comment (fivetran-reneeli) [reply README.md:1]: Merged that in and updated some of the language/structure.

**Comment by fivetran-reneeli** [comment]
Thanks @fivetran-joemarkiewicz  this is ready for re-review. I updated with the validation test results. Actually, as expected, there was one test failure on `consistency_user_campaign`, due to the addition of the new `first_open_or_click_event_at` field.

File: models/iterable__events.sql
Comment (fivetran-joemarkiewicz) [inline models/iterable__events.sql:3]: Based on our internal discussion, this change can be reverted. We just will need to make sure we update the seed data to replicate a realistic environment.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli thanks for working through these changes! I have a few comments before this is ready for approval. Let me know if you have questions and once this is ready for final review.

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:4]: updated

File: integration_tests/dbt_project.yml
Comment (fivetran-reneeli) [reply integration_tests/dbt_project.yml:23]: nope used it to test and figured to just comment it out for future use when testing, but I can remove it

File: models/iterable__events.sql
Comment (fivetran-reneeli) [reply models/iterable__events.sql:3]: updated and updated seed!

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:9]: updated

File: models/iterable__events.sql
Comment (fivetran-reneeli) [reply models/iterable__events.sql:20]: updated here and elsewhere the variable is mentioned

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:10]: yes added!

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-reneeli this is really close, just a few small changes before approval. Let me know once their updated. Thanks!

File: README.md
Comment (fivetran-reneeli) [reply README.md:170]: makes sense! thank you

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:4]: added!

File: models/iterable.yml
Comment (fivetran-reneeli) [reply models/iterable.yml:159]: added

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-reneeli changes look good to me! Just a small CHANGELOG edit request and a notice that I made a small README change to close a codeblock. Once the CHANGELOG edit is made this is ready for release review.

File: README.md
Comment (fivetran-reneeli) [reply README.md:188]: oh gotcha, thank you!

File: CHANGELOG.md
Comment (fivetran-avinash) [inline CHANGELOG.md:13]: Looks like this section started with Justin's PR, can we call that out in this line. 

```suggestion
- Added a section in the [README]((https://github.com/fivetran/dbt_iterable/blob/main/README.md#pivoting-out-event-metrics)) about the `iterable__event_metrics` variable and how to use it to specify which event metrics to pivot out. ([PR #49](https://github.com/fivetran/dbt_iterable/pull/49))
```

File: packages.yml
Comment (fivetran-avinash) [reply packages.yml:7]: Same

File: CHANGELOG.md
Comment (fivetran-avinash) [inline CHANGELOG.md:9]:
```suggestion
- In order to ensure no issues, a `--full-refresh` is required after upgrading.
```

File: integration_tests/seeds/event_data.csv
Comment (fivetran-avinash) [inline integration_tests/seeds/event_data.csv:1]: @fivetran-reneeli We should probably have the seed files match on [both source](https://github.com/fivetran/dbt_iterable_source/pull/39/files#diff-300565a6c128bc16b60b3661cccd197e268805449eb3504ed1b05129bd6825e8) and transform packages. Can you update one or the other? Not sure what works best here for more robust testing.

File: CHANGELOG.md
Comment (fivetran-avinash) [inline CHANGELOG.md:15]:
```suggestion
- Modified the `event` seed data to more accurately represent real-life data, with a unique `_fivetran_id` for each `campaign_id`, `created_at`, and `event_name`.
```

File: CHANGELOG.md
Comment (fivetran-avinash) [inline CHANGELOG.md:16]: @fivetran-reneeli Looks like you made the same documentation update in `iterable.yml` that you did in the source yml files, should this be added?

```suggestion
## Documentation Update
- Updates the descriptions of timestamp-based fields. Previously they were described as milliseconds since epoch time, but they should be standard timestamps.

```

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-reneeli Nice work! A few comments before approval.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-reneeli  Oops, something went wonky here--had two tabs open with comments/suggestions and submitted both. Let me know if you have any questions.

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:13]: Thanks for catching !

File: integration_tests/seeds/event_data.csv
Comment (fivetran-reneeli) [reply integration_tests/seeds/event_data.csv:1]: Thanks @fivetran-avinash , I believe they're the same now

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-reneeli Ready to merge!

**Pull request merged**
